crates/ide-assists/src/handlers/unmerge_use.rs

```diff
@@ -44,16 +44,24 @@ pub(crate) fn unmerge_use(acc: &mut Assists, ctx: &AssistContext<'_>) -> Option<
 
     let target = tree.syntax().text_range();
     acc.add(AssistId("unmerge_use", AssistKind::RefactorRewrite), label, target, |builder| {
-        let new_use = make::use_(
+
+        let make = SyntaxFactory::new();
+
+        let mut editor = builder.make_editor(&new_parent);
+
+        let new_use = make.use_(
             use_.visibility(),
-            make::use_tree(path, tree.use_tree_list(), tree.rename(), tree.star_token().is_some()),
+            make.use_tree(path, tree.use_tree_list(), tree.rename(), tree.star_token().is_some()),
         )
         .clone_for_update();
 
         tree.remove();
-        ted::insert(Position::after(use_.syntax()), new_use.syntax());
+        editor.insert(Position::after(use_.syntax()), new_use.syntax());
 
         builder.replace(old_parent_range, new_parent.to_string());
```

Comment:
This should use `editor.replace(...)`

Reply:
Similar to the question above. It appears the `old_parent_range` and `new_parent.to_string()` arguments are invalid for the `editor.replace` method as is.

Reply:
Right, we'll want to pass the corresponding `SyntaxNode`s instead. We don't need the _range_ of the old parent, but the old parent itself. Similarly, we don't need the stringified new parent, but the new parent itself.

Reply:
The original bug seems to be fixed but I am getting a new error. This one appears similar to the other 'immutable tree' error I was having in `add_braces`.

failures:

---- handlers::unmerge_use::tests::unmerge_glob_import stdout ----
thread 'handlers::unmerge_use::tests::unmerge_glob_import' panicked at /home/codespace/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rowan-0.15.15/src/cursor.rs:824:9:
immutable tree: use std::fmt::*;
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- handlers::unmerge_use::tests::unmerge_indented_use_item stdout ----
thread 'handlers::unmerge_use::tests::unmerge_indented_use_item' panicked at /home/codespace/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rowan-0.15.15/src/cursor.rs:824:9:
immutable tree: use std::fmt::Display as Disp;

---- handlers::unmerge_use::tests::unmerge_nested_use_item stdout ----
thread 'handlers::unmerge_use::tests::unmerge_nested_use_item' panicked at /home/codespace/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rowan-0.15.15/src/cursor.rs:824:9:
immutable tree: use foo::bar::baz::qux;

---- handlers::unmerge_use::tests::unmerge_renamed_use_item stdout ----
thread 'handlers::unmerge_use::tests::unmerge_renamed_use_item' panicked at /home/codespace/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rowan-0.15.15/src/cursor.rs:824:9:
immutable tree: use std::fmt::Display as Disp;

---- handlers::unmerge_use::tests::unmerge_use_item stdout ----
thread 'handlers::unmerge_use::tests::unmerge_use_item' panicked at /home/codespace/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rowan-0.15.15/src/cursor.rs:824:9:
immutable tree: use std::fmt::Display;

---- handlers::unmerge_use::tests::unmerge_use_item_on_self stdout ----
thread 'handlers::unmerge_use::tests::unmerge_use_item_on_self' panicked at /home/codespace/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rowan-0.15.15/src/cursor.rs:824:9:
immutable tree: use std::process;

---- handlers::unmerge_use::tests::unmerge_use_item_with_visibility stdout ----
thread 'handlers::unmerge_use::tests::unmerge_use_item_with_visibility' panicked at /home/codespace/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rowan-0.15.15/src/cursor.rs:824:9:
immutable tree: pub use std::fmt::Display;

---- tests::generated::doctest_unmerge_use stdout ----
thread 'tests::generated::doctest_unmerge_use' panicked at /home/codespace/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rowan-0.15.15/src/cursor.rs:824:9:
immutable tree: use std::fmt::Display;


failures:
    handlers::unmerge_use::tests::unmerge_glob_import
    handlers::unmerge_use::tests::unmerge_indented_use_item
    handlers::unmerge_use::tests::unmerge_nested_use_item
    handlers::unmerge_use::tests::unmerge_renamed_use_item
    handlers::unmerge_use::tests::unmerge_use_item
    handlers::unmerge_use::tests::unmerge_use_item_on_self
    handlers::unmerge_use::tests::unmerge_use_item_with_visibility
    tests::generated::doctest_unmerge_use

test result: FAILED. 2323 passed; 8 failed; 1 ignored; 0 measured; 0 filtered out; finished in 1.84s

Reply:
I think you're probably missing a `clone_for_update`. If you push your changes I can try to help debug.